internal/pkg/manifests/store/builder.go

```diff
@@ -0,0 +1,377 @@
+package store
+
+import (
+	"fmt"
+
+	monitoringthanosiov1alpha1 "github.com/thanos-community/thanos-operator/api/v1alpha1"
```

Comment:
I think ideally, we would avoid any imports of the api package into here because it will prevent us from easily adding new versions or for example swapping out the builders to use something like we did in rhobs/config. translation between openapi spec and the types in this package should happen at the ctrl layer. does that make sense?

it might not work that trivially in code but that was the underlying idea